sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/models/ShareFileHandleAccessRights.java

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.storage.file.share.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+/** Access rights of the access policy. */
+public enum ShareFileHandleAccessRights {
```

Comment:
Is this actually code generated? If not, we can have a Swagger transform to generate `AccessRight` with this class name so we can remove the conversion layer.

Reply:
This was a handwritten class, the generated AccessRight.java class lives in the implementation folder.